tests/unit/asio_service_test.cxx

```diff
@@ -657,6 +657,32 @@ int response_hint_test(bool with_meta) {
     CHK_OK( s2.getTestSm()->isSame( *s1.getTestSm() ) );
     CHK_OK( s3.getTestSm()->isSame( *s1.getTestSm() ) );
 
+    _msg("enable batch size hint with negative value\n");
+    for (RaftAsioPkg* ee: pkgs) {
+        ee->getTestSm()->set_next_batch_size_hint_in_bytes(-1);
+    }
+
+    TestSuite::sleep_sec(1, "wait peer's hint size info refreshed in leader side");
+
+    // With negative hint size, append_entries will timeout due to
+    // raft server can not commit. Set timeout to a small value.
+    raft_params params = s1.raftServer->get_current_params();
+    params.with_client_req_timeout(1000);
+    s1.raftServer->update_params(params);
+
+    for (size_t ii=0; ii<3; ++ii) {
+        std::string msg_str = "3rd_" + std::to_string(ii);
+        ptr<buffer> msg = buffer::alloc(sizeof(uint32_t) + msg_str.size());
+        buffer_serializer bs(msg);
+        bs.put_str(msg_str);
+        s1.raftServer->append_entries( {msg} );
+    }
+    TestSuite::sleep_sec(1, "wait for replication but actually no replication happen");
```

Comment:
If you temporarily increase this sleep time much longer, you can see that Raft is overwhelming CPU, because `need_to_catchup` is `true`. The leader is continuously sending meaningless messages to followers whose hint is `-1`. I think if hint is `-1`, `need_to_catchup` flag should be `false` so that the leader sends heartbeats only.
```
2020-08-05T12:22:34.782_025-07:00 [3b34] [DEBG] Receive a append_entries_response message from peer 3 with Result=1, Term=1, NextIndex=204  [raft_server.cxx:681, handle_peer_resp()]
2020-08-05T12:22:34.782_029-07:00 [3b34] [TRAC] src: 3, dst: 1, resp->get_term(): 1 [raft_server.cxx:684, handle_peer_resp()]
2020-08-05T12:22:34.782_034-07:00 [3b34] [TRAC] handle append entries resp (from 3), resp.get_next_idx(): 204   [handle_append_entries.cxx:800, handle_append_entries_resp()]
2020-08-05T12:22:34.782_038-07:00 [3b34] [TRAC] peer 3 batch size hint: -1 bytes    [handle_append_entries.cxx:803, handle_append_entries_resp()]
2020-08-05T12:22:34.782_041-07:00 [3b34] [TRAC] peer 3, prev matched idx: 203, new matched idx: 203 [handle_append_entries.cxx:815, handle_append_entries_resp()]
2020-08-05T12:22:34.782_049-07:00 [3b34] [TRAC] quorum idx 1, 206 203 203   [handle_append_entries.cxx:956, get_expected_committed_log_idx()]
2020-08-05T12:22:34.782_054-07:00 [3b34] [TRAC] local log idx 206, target_commit_idx 203, quick_commit_index_ 203, state_->get_commit_idx() 203 [handle_commit.cxx:59, commit()]
2020-08-05T12:22:34.782_057-07:00 [3b34] [DEBG] reqeust append entries need to catchup, p 3 [handle_append_entries.cxx:924, handle_append_entries_resp()]
2020-08-05T12:22:34.782_062-07:00 [3b34] [TRAC] send request to 3   [handle_append_entries.cxx:154, request_append_entries()]
2020-08-05T12:22:34.782_066-07:00 [3b34] [TRAC] last_log_idx: 203, starting_idx: 191, cur_nxt_idx: 207  [handle_append_entries.cxx:288, create_append_entries_req()]
2020-08-05T12:22:34.782_069-07:00 [3b34] [DEBG] last sent log (204) to peer 3 is not applied, cnt 50843 [handle_append_entries.cxx:308, create_append_entries_req()]
2020-08-05T12:22:34.782_071-07:00 [3b34] [DEBG] reduce end_idx 207 -> 205   [handle_append_entries.cxx:312, create_append_entries_req()]
2020-08-05T12:22:34.782_077-07:00 [3b34] [TRAC] adjusted end_idx due to batch size hint: 205 -> 204 [handle_append_entries.cxx:386, create_append_entries_req()]
2020-08-05T12:22:34.782_081-07:00 [3b34] [DEBG] append_entries for 3 with LastLogIndex=203, LastLogTerm=1, EntriesLength=0, CommitIndex=203, Term=1, peer_last_sent_idx 204 [handle_append_entries.cxx:394, create_append_entries_req()]
2020-08-05T12:22:34.782_084-07:00 [3b34] [TRAC] EMPTY PAYLOAD   [handle_append_entries.cxx:396, create_append_entries_req()]
2020-08-05T12:22:34.782_092-07:00 [3b34] [TRAC] send req 1 -> 3, type append_entries_request    [peer.cxx:43, send_req()]
2020-08-05T12:22:34.782_123-07:00 [3b34] [TRAC] sent    [handle_append_entries.cxx:224, request_append_entries()]
```

Reply:
I fixed it by myself and looks good. Let me merge it thanks.